Mixmatch/streetview_dataset/parse_TFRecord.ipynb

```diff
@@ -249,6 +252,269 @@
+    "OUTPUT_TRAIN_RECORD_FILENAME = 'streetview_v3_64-train.tfrecord'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### streetview_v3-test\n",
+    "This test set uses all TC11 images as positive cases, and handpick UCF images with confidence<0.2 as negative cases."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# filter the dataset with images lower than the threshold, and store the image names in a list. These images will be handpicked to be used as negative examples in the test set.\n",
+    "def filter_image(parsed_image_dataset, folder_path, threshold):\n",
+    "    res = []\n",
+    "    \n",
+    "    for image_features in parsed_image_dataset:\n",
+    "        img_name, confidence, bbox = parse_labels(image_features)\n",
+    "\n",
+    "        if img_name:\n",
+    "            target = []\n",
+    "            if confidence.size==0 or (confidence.size>0 and max(confidence)<threshold):\n",
+    "                res.append(img_name)\n",
+    "    return res\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Filter UCF images and copy files for furthur handpicking test set.\n",
+    "res = []\n",
+    "for i in range(0.9 * INPUT_RECORD_CNT, INPUT_RECORD_CNT):\n",
+    "    file_name = \"./streetlearn_detections_tfexample-\" + str(i).zfill(5) + \"-of-01000.tfrecord\"\n",
+    "    parsed_image_dataset = read_TFRecord(os.path.join(INPUT_RECORD_PATH, file_name))\n",
+    "    res.extend(filter_image(parsed_image_dataset, INPUT_UCF_IMG_DIR, 0.2))\n",
+    "\n",
+    "for file in res:\n",
+    "    # The format fo the image_name is XXXXXX_Y.jpg, the Y represents the view of the image.\n",
+    "    view = img_name.split('.')[0][-1]\n",
+    "    # ignore sky images\n",
+    "    if view != '5' and view != '0':\n",
+    "        copyfile(INPUT_UCF_IMG_DIR + file, OUTPUT_UCF_IMG_DIR + file)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Build streetview_v3_64-test. Write test tfrecord from TC11 dataset and filtered UCF\n",
+    "test_writer = tf.io.TFRecordWriter(OUTPUT_RECORD_PATH + OUTPUT_TEST_RECORD_FILENAME)\n",
+    "test_image_path = os.path.join(INPUT_TC11_IMG_DIR, 'img')\n",
+    "write_path2tfrecord(test_image_path, 1, test_writer)\n",
+    "write_path2tfrecord(INPUT_UCF_FILTER_IMG_DIR, 0, test_writer)\n",
+    "test_writer.close()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### streetview_v3-train (Unbiased UCF training set)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def write_tfrecord_ucf_v3(parsed_image_dataset, folder_path, writer, balance_threshold):\n",
```

Comment:
Please merge all the three functions into a single one, because there are a lot of common logic.

An ideal function interface can be like:
```
def generate_tfexamples_from_detections(detection_results, image_dir_path, only_keep_top_confidence, exclude_top_camera):
    // generate positive and negative sets separately
    ...
    return positive_examples, negative_examples

def write_tfexample_to_tfrecord(postive_examples, negative_examples, pos_to_neg_ratio, writer):
    num_pos = ...
    num_neg = ...
    output = [positive_examples[:num_pos], negative_examples[:num_neg]]  // you may need to figure out the right syntax here.
    writer.write(output)
```
And I believe this interface should be reusable across all the data sets by providing different params.

Reply:
Create both suggested functions and merge all three functions.